socorro/external/es/supersearch.py

```diff
@@ -137,8 +142,17 @@ def _format(aggregation):
                 return aggregation
 
             for i, bucket in enumerate(aggregation["buckets"]):
+                term = bucket.get("key_as_string")
+                if "key_as_string" in bucket:
+                    term = bucket["key_as_string"]
+                    if term in ("true", "false"):
+                        # Restore es 1.4 format for boolean terms as string
+                        term = term[:1].upper()
+                else:
+                    term = bucket["key"]
```

Comment:
As i understand it the difference is externally visible, so I want to keep the behavior the same to minimize migration risk.

Reply:
I changed it to this, which I think should get the correct behavior:
```python
                if bucket["key"] in (True, False):
                    # Restore es 1.4 format for boolean terms as string
                    term = "T" if bucket["key"] else "F"
                elif "key_as_string" in bucket:
                    term = bucket["key_as_string"]
                else:
                    term = bucket["key"]
```

Reply:
I was wrong, turns out `0.0 in (True, False)` is True, I had to change this to use `if isinstance(bucket["key"], bool):` to fix the topcrashers report

Reply:
Oh, right... that's because in Python, False is equivalent to 0.

```
> python
Python 3.12.6 (main, Sep  9 2024, 22:11:19) [Clang 18.1.8 ] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> False == 0
True
>>> False == 0.0
True
>>> False + False
0
>>> True + True
2
```

Fun fun fun.